policies.go

```diff
@@ -519,7 +521,8 @@ func (t *tokenAwareHostPolicy) AddHosts(hosts []*HostInfo) {
 
 func (t *tokenAwareHostPolicy) RemoveHost(host *HostInfo) {
 	t.mu.Lock()
-	if t.hosts.remove(host.ConnectAddress()) {
+	connAddr, _ := host.ConnectAddress()
+	if t.hosts.remove(connAddr) {
```

Comment:
Same as above but needs to call `RemoveHost` on the fallback policy instead of returning right away